File: 백준/Silver/9613. GCD 합/GCD 합.py
import sys

def gcd(x, y):
  while x:
    x, y = y % x, x

  return y

t = int(input())

for i in range(t):
  num = list(map(int, sys.stdin.readline().split()))
  n = num.pop(0)
  sum = 0

  for i in range(n - 1):
    for j in range(i+1, n):
      sum += gcd(num[i], num[j])

  print(sum)

File: 백준/Silver/9613. GCD 합/GCD 합.cc
#include <iostream>
using namespace std;

int gcd(int x, int y) {
  while(x) {
    int tmp = x;
    x = y % x;
    y = tmp;
  }
  
  return y;
}

int main() {
  int t, n;
  cin >> t;

  for (int i = 0; i < t; i++) {
    cin >> n;
    int num[n];
    long long sum = 0;

    for (int j = 0; j < n; j++)
      cin >> num[j];

    for (int j = 0; j < n-1; j++) {
      for (int k = j+1; k < n; k++) {
        sum += gcd(num[j], num[k]);
      }
    }

    cout << sum << "\n";
  }
}